### DISCUSSION TITLE
Lighthouse authorization with token/data in session storage.

### DISCUSSION THREAD

**Discussion by llwch** [GoogleChrome/lighthouse#12319] [Q&A]
Hi. 
I have web application with login process that store token and some user data inside browser session storage. 
Is it possible to run some auth code (like filling session storage with required data) during lighthouse run?

Or the only way - wait for Flow Support (Fraggle Rock) #11313 where the same tab can be used?

Thanks!

**Comment by connorjclark** [comment]
Easiest approach is to wait for Fraggle Rock work to be done, but unclear how long :) 

Have you seen https://github.com/GoogleChrome/lighthouse/blob/master/docs/authenticated-pages.md ?

**Reply by llwch** [reply]
Yes, sure, I saw that page. Also using such options:
`const lightHouseOpt = {
      port: port,
      output: 'json',
      onlyCategories: ['performance'],
      disableStorageReset: true,
      logLevel: 'info',
    }`
But disableStorageReset will not help with session storage, that's why lighthouse always opens a new tab on auth page. I've tried to execute a script that calls for API, takes token and other info, and paste it into session storage. And it works well for ordinary login, but I have no ideas how to use it inside lighthouse execution. Even tried something like 
` browser.on('targetcreated', function () {
    //some code
  });` but with no success.
The easiest way - no use local storage, but customer doesn't want to do it due to security reasons.
